shell/chart/monitoring/index.vue

```diff
@@ -176,6 +176,10 @@ export default {
       const selector =
         prometheusSpec?.storageSpec?.volumeClaimTemplate?.spec?.selector;
 
+      // This works for UI editor installation
+      // However, it doesn't work for yaml editor installation
+      // Global values later merged again in charts/install.vue addGlobalValuesTo()
+      // We still need to remove the global values from charts/install.vue addGlobalValuesTo()
```

Comment:
After trawling through the monitoring code, I think this original solution is just not great. That's hardly your fault but I think https://github.com/rancher/dashboard/issues/8723 can be fixed without adding further complexity. The empty selector is initialized when the 'persistent storage for prometheus' checkbox is clicked:
https://github.com/rancher/dashboard/blob/d91591ceb70e22e5de497c2eb8db5f0a284f9ede/shell/chart/monitoring/prometheus/index.vue#L149-L160

Rather than the original solution of always adding something that breaks some storage providers then removing it before save, I think we could just stop initializing the `volumeClaimTemplate` this way (ie delete line 157 linked above) and update `matchChanged` in the same file to delete `volumeClaimTemplate.spec.selector` if all selectors are cleared from the UI https://github.com/rancher/dashboard/blob/d91591ceb70e22e5de497c2eb8db5f0a284f9ede/shell/chart/monitoring/prometheus/index.vue#L173-L184

Would become something like:
```
    matchChanged(expressions) {
      const { matchLabels, matchExpressions } = simplify(expressions);
      const storageSpec = this.value.prometheus.prometheusSpec.storageSpec.volumeClaimTemplate.spec;

      if (storageSpec?.selector && isEmpty(matchLabels) && isEmpty(matchExpressions)) {
        delete storageSpec.selector;
      } else {
        this.$set(storageSpec, 'selector', { matchLabels, matchExpressions });
      }
    },
```